docker/zarr-output/create_zarr_optimus.py

```diff
@@ -160,49 +160,34 @@ def add_cell_metrics(
 
     # Read the csv input files
     metrics_df = pd.read_csv(metrics_file, dtype=str)
-    emptydrops_df = pd.read_csv(emptydrops_file, dtype=str)
+ 
+    if add_emptydrops_results == 'yes':
+       emptydrops_df = pd.read_csv(emptydrops_file, dtype=str)
 
     # Check that input is valid
     if metrics_df.shape[0] == 0 or metrics_df.shape[1] == 0:
         logging.error("Cell metrics table is not valid")
         raise ValueError()
-    if emptydrops_df.shape[0] == 0 or emptydrops_df.shape[1] == 0:
-        logging.error("EmptyDrops table is not valid")
-        raise ValueError()
+
+    if add_emptydrops_results == 'yes':
+        if emptydrops_df.shape[0] == 0 or emptydrops_df.shape[1] == 0:
+            logging.error("EmptyDrops table is not valid")
+            raise ValueError()
 
     # Rename cell columns for both datasets to cell_id
-    emptydrops_df = emptydrops_df.rename(columns={"CellId": "cell_id"})
+    if add_emptydrops_results == 'yes':
+       emptydrops_df = emptydrops_df.rename(columns={"CellId": "cell_id"})
```

Comment:
It appears that the definition of emptydrops_df is after the validation on line 173